### PR TITLE
Enrollment location map

### PR DESCRIPTION
**Pull request by gigxz**
## Description

Add Enrollment Location Map to enrollment dashboard.

Issue: https://github.com/open-path/Green-River/issues/7016
Depends on hmis-warehouse PR: https://github.com/greenriver/hmis-warehouse/pull/4993
Includes https://github.com/greenriver/hmis-frontend/pull/992

## Type of change
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

**Comment by gigxz** [comment]
Still need to adjust file structure https://github.com/greenriver/hmis-frontend/pull/992#discussion_r1894022739 

edit: done with https://github.com/greenriver/hmis-frontend/pull/997/commits/ac9c1272af3c23471ea68314fe1d20e3818e737e

**Comment by gigxz** [comment]
Flipping between CLS and Location Map tabs on mobile is not optimal, you need to vertical scroll the middle area which has a small height
Screenshot 2024-12-23 at 4 01 28 PM

File: src/modules/enrollment/components/pages/EnrollmentLocationMapPage.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { clientBriefName } from '@/modules/hmis/hmisUtil';
+
+const EnrollmentLocationMapPage = () => {
+  const { enrollment } = useEnrollmentDashboardContext();
+
+  if (!enrollment) return <NotFound />;
+
+  return (
+    <TitleCard title='Location Map' headerVariant='border'>
+      <Box sx={{ p: 2 }}>
```
Comment (ttoomey) [inline src/modules/enrollment/components/pages/EnrollmentLocationMapPage.tsx]: minor thing, I might take the padding off to make better use of horizontal space on mobile

File: src/components/elements/maps/SingleGeolocationMap.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  return (
+    <BaseMapContainer
+      key={JSON.stringify(coordinates)}
+      height={height}
+      center={[coordinates.latitude, coordinates.longitude]}
+      zoom={16}
+      dragging={false}
+      scrollWheelZoom={false}
+      doubleClickZoom
+      touchZoom
```
Comment (ttoomey) [inline src/components/elements/maps/SingleGeolocationMap.tsx]: minor thing,  some of these props seem to be on both single and multi location maps. Perhaps they should be default on the BaseMapContainer

File: src/components/elements/maps/MultiGeolocationMap.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  // find bounds for the map
+  const bounds = useMemo(() => {
+    const latLngExpressions: LatLngExpression[] = geolocations.map((loc) =>
+      locationToLatLngExpression(loc)
+    );
+    return L.latLngBounds(latLngExpressions);
+  }, [geolocations]);
+
+  return (
+    <BaseMapContainer
```
Comment (ttoomey) [inline src/components/elements/maps/MultiGeolocationMap.tsx:42]: The SingleGeolocationMap gets a key prop, I assume to force re-render. Do we need anything like that here?

**Review by ttoomey — APPROVED** [review]
Maps look great!

File: src/modules/enrollment/components/pages/EnrollmentLocationMapPage.tsx (hunk 1 above)
Comment (gigxz) [reply src/modules/enrollment/components/pages/EnrollmentLocationMapPage.tsx]: good idea, updated!

File: src/components/elements/maps/SingleGeolocationMap.tsx (hunk 2 above)
Comment (gigxz) [reply src/components/elements/maps/SingleGeolocationMap.tsx]: good idea, moved some to BaseMapContainer

File: src/components/elements/maps/MultiGeolocationMap.tsx (hunk 3 above)
Comment (gigxz) [reply src/components/elements/maps/MultiGeolocationMap.tsx:42]: The key was not necessary on the SingleGeolocationMap map, I removed it. I think it was from a previous iteration but I confirmed it renders as expected without. THanks!

**Pull request merged**